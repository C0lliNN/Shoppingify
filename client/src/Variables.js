export const COLORS = {
  primary: '#F9A109',
  secondary: '#56CCF2',
  danger: '#EB5757',
  white: '#FFFFFF',
  gray_1: '#E0E0E0',
  gray_2: '#BDBDBD',
  black_1: '#000000',
  black_2: '#34333A',
};

export const FONT_FAMILY = 'Quicksand';
export const FONT_SIZE_1 = '14px';

export const BORDER_RADIUS_1 = '12px;';
export const BORDER_RADIUS_2 = '24px;';
