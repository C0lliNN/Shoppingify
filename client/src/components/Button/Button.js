import PropTypes from 'prop-types';
import React from 'react';
import styles from './Button.module.css';

export default function Button(props) {
  const classes = [styles.Button];

  switch (props.type) {
    case 'primary':
      classes.push(styles.Primary);
      break;
    case 'success':
      classes.push(styles.Success);
      break;
    case 'danger':
      classes.push(styles.Danger);
      break;
    default:
      classes.push(styles.Primary);
  }

  return (
    <button onClick={props.onClick} className={classes.join(' ')}>
      {props.children}
    </button>
  );
}

Button.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  onClick: PropTypes.func,
  type: PropTypes.oneOf(['primary', 'success', 'danger']),
};
