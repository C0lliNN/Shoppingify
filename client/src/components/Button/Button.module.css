.Button {
  background-color: #303030;
  outline: none;
  border: none;
  padding: 10px 20px;
  border-radius: 5px;
  font-size: 1em;
  color: white;
  cursor: pointer;
}

.Button:hover {
  background-color: #2b2b2b;
}

.Primary {
  background-color: rgb(0, 119, 255);
}

.Primary:hover {
  background-color: rgb(4, 108, 226);
}

.Success {
  background-color: rgb(14, 156, 90);
}
.Success:hover {
  background-color: rgb(14, 138, 80);
}

.Danger {
  background-color: rgb(255, 0, 64);
}

.Danger:hover {
  background-color: rgb(231, 0, 58);
}
