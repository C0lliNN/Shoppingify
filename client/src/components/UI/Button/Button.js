import styled from 'styled-components';
import {
  COLORS,
  BORDER_RADIUS_1,
  BORDER_RADIUS_2,
  FONT_FAMILY,
  FONT_SIZE_2,
  BUTTON_TYPES,
} from '../../../helpers/style-constants';
import PropTypes from 'prop-types';

const Button = styled.button`
  ${(props) => {
    switch (props.btnType) {
      case 'raised': {
        return `
          background-color: ${COLORS[props.variant]}; 
          border-color: ${COLORS[props.variant]};
          border-radius: ${BORDER_RADIUS_1}; 
          color: ${COLORS.white}; 
          padding: 20px;
        `;
      }
      case 'outlined': {
        return `
          background-color: transparent;
          border-radius: ${BORDER_RADIUS_2}; 
          border-color: ${COLORS[props.variant]};
          color: ${COLORS[props.variant]};
          padding: 6px 15px;
        `;
      }
      case 'flat': {
        return `
          background-color: transparent;
          border-radius: ${BORDER_RADIUS_1}; 
          color: ${COLORS.black_2};
          border-color: transparent;
          padding: 20px;
        `;
      }
      default:
        return `
          background-color: ${COLORS[props.variant]}; 
          border-color: ${COLORS[props.variant]};
          border-radius: ${BORDER_RADIUS_1}; 
          color: ${COLORS.white}; 
          padding: 20px;
        `;
    }
  }}
  border-width: 2px;
  border-style: solid;
  font-size: ${FONT_SIZE_2};
  font-family: ${FONT_FAMILY};
  font-weight: bold;
  outline: none;
  cursor: pointer;
  &:disabled {
    ${(props) => {
      switch (props.btnType) {
        case 'raised': {
          return `
            background-color: ${COLORS.gray_2};
          `;
        }
        case 'outlined': {
          return `
            color: ${COLORS.gray_2}; 
          `;
        }
        case 'flat': {
          return `
            color: ${COLORS.gray_2}
          `;
        }
        default:
          return `
        background-color: ${COLORS.gray_2};
        `;
      }
    }}
  
    border-color: ${COLORS.gray_2};  
    cursor: default;
  }
`;

Button.propTypes = {
  btnType: PropTypes.oneOf(BUTTON_TYPES).isRequired,
  variant: PropTypes.oneOf(Object.keys(COLORS)),
};

export default Button;
