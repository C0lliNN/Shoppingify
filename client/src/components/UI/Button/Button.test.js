import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import Button from './Button';
import { COLORS } from '../../../Variables';

describe('<Button/>', () => {
  let errorMessage = null;
  beforeEach(() => {
    console.error = jest.fn((message) => {
      errorMessage = message;
    });
  });
  it('should console.error if the btnType is undefined', () => {
    render(<Button variant="primary">Button</Button>);

    expect(console.error).toHaveBeenCalled();
    expect(errorMessage).toMatch(/btnType.*required.*/i);
  });
  it('should console.error if btnType is undefined', () => {
    render(
      <Button btnType="invalid-btnType" variant="primary">
        Button
      </Button>
    );

    expect(console.error).toHaveBeenCalled();
    expect(errorMessage).toMatch(/invalid.*btnType.*/i);
  });
  it('should console.error if the variant is undefined', () => {
    render(<Button btnType="flat">Button</Button>);

    expect(console.error).toHaveBeenCalled();
    expect(errorMessage).toMatch(/variant.*required.*/i);
  });
  it('should console.error if variant is undefined', () => {
    render(
      <Button btnType="raised" variant="invalid-variant">
        Button
      </Button>
    );

    expect(console.error).toHaveBeenCalled();
    expect(errorMessage).toMatch(/invalid.*variant.*/i);
  });
  it('should render correctly when the type is raised and the variant is valid', () => {
    Object.keys(COLORS).forEach((color) => {
      const { container } = render(
        <Button btnType="raised" variant={color}>
          Button
        </Button>
      );
      expect(container).toMatchSnapshot();
      cleanup();
    });
  });
  it('should render correctly when the type is flat and the variant is valid', () => {
    Object.keys(COLORS).forEach((color) => {
      const { container } = render(
        <Button btnType="flat" variant={color}>
          Button
        </Button>
      );
      expect(container).toMatchSnapshot();
      cleanup();
    });
  });
  it('should execute the function passed through onClick when the button is clicked', () => {
    const handler = jest.fn();

    render(<Button onClick={handler}>Button</Button>);
    screen.queryByText('Button').click();

    expect(handler).toHaveBeenCalled();
  });
});
