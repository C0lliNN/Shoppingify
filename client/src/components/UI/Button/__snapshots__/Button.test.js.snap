// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> should render correctly when the type is flat 1`] = `
.c0 {
  background-color: transparent;
  border-radius: 12px;
  color: #34333A;
  border-color: transparent;
  padding: 5px;
  border-width: 2px;
  border-style: solid;
  font-size: 14px;
  font-family: Quicksand;
  font-weight: bold;
  outline: none;
  cursor: pointer;
}

.c0:disabled {
  color: #BDBDBD   border-color:#BDBDBD;
  cursor: default;
}

<div>
  <button
    class="c0"
  >
    Button
  </button>
</div>
`;

exports[`<Button/> should render correctly when the type is outlined and the variant is valid 1`] = `
.c0 {
  background-color: transparent;
  border-radius: 24px;
  border-color: #F9A109;
  color: #F9A109;
  padding: 6px 15px;
  border-width: 2px;
  border-style: solid;
  font-size: 14px;
  font-family: Quicksand;
  font-weight: bold;
  outline: none;
  cursor: pointer;
}

.c0:disabled {
  color: #BDBDBD;
  border-color: #BDBDBD;
  cursor: default;
}

<div>
  <button
    class="c0"
  >
    Button
  </button>
</div>
`;

exports[`<Button/> should render correctly when the type is outlined and the variant is valid 2`] = `
.c0 {
  background-color: transparent;
  border-radius: 24px;
  border-color: #56CCF2;
  color: #56CCF2;
  padding: 6px 15px;
  border-width: 2px;
  border-style: solid;
  font-size: 14px;
  font-family: Quicksand;
  font-weight: bold;
  outline: none;
  cursor: pointer;
}

.c0:disabled {
  color: #BDBDBD;
  border-color: #BDBDBD;
  cursor: default;
}

<div>
  <button
    class="c0"
  >
    Button
  </button>
</div>
`;

exports[`<Button/> should render correctly when the type is outlined and the variant is valid 3`] = `
.c0 {
  background-color: transparent;
  border-radius: 24px;
  border-color: #EB5757;
  color: #EB5757;
  padding: 6px 15px;
  border-width: 2px;
  border-style: solid;
  font-size: 14px;
  font-family: Quicksand;
  font-weight: bold;
  outline: none;
  cursor: pointer;
}

.c0:disabled {
  color: #BDBDBD;
  border-color: #BDBDBD;
  cursor: default;
}

<div>
  <button
    class="c0"
  >
    Button
  </button>
</div>
`;

exports[`<Button/> should render correctly when the type is outlined and the variant is valid 4`] = `
.c0 {
  background-color: transparent;
  border-radius: 24px;
  border-color: #FFFFFF;
  color: #FFFFFF;
  padding: 6px 15px;
  border-width: 2px;
  border-style: solid;
  font-size: 14px;
  font-family: Quicksand;
  font-weight: bold;
  outline: none;
  cursor: pointer;
}

.c0:disabled {
  color: #BDBDBD;
  border-color: #BDBDBD;
  cursor: default;
}

<div>
  <button
    class="c0"
  >
    Button
  </button>
</div>
`;

exports[`<Button/> should render correctly when the type is outlined and the variant is valid 5`] = `
.c0 {
  background-color: transparent;
  border-radius: 24px;
  border-color: #E0E0E0;
  color: #E0E0E0;
  padding: 6px 15px;
  border-width: 2px;
  border-style: solid;
  font-size: 14px;
  font-family: Quicksand;
  font-weight: bold;
  outline: none;
  cursor: pointer;
}

.c0:disabled {
  color: #BDBDBD;
  border-color: #BDBDBD;
  cursor: default;
}

<div>
  <button
    class="c0"
  >
    Button
  </button>
</div>
`;

exports[`<Button/> should render correctly when the type is outlined and the variant is valid 6`] = `
.c0 {
  background-color: transparent;
  border-radius: 24px;
  border-color: #BDBDBD;
  color: #BDBDBD;
  padding: 6px 15px;
  border-width: 2px;
  border-style: solid;
  font-size: 14px;
  font-family: Quicksand;
  font-weight: bold;
  outline: none;
  cursor: pointer;
}

.c0:disabled {
  color: #BDBDBD;
  border-color: #BDBDBD;
  cursor: default;
}

<div>
  <button
    class="c0"
  >
    Button
  </button>
</div>
`;

exports[`<Button/> should render correctly when the type is outlined and the variant is valid 7`] = `
.c0 {
  background-color: transparent;
  border-radius: 24px;
  border-color: #828282;
  color: #828282;
  padding: 6px 15px;
  border-width: 2px;
  border-style: solid;
  font-size: 14px;
  font-family: Quicksand;
  font-weight: bold;
  outline: none;
  cursor: pointer;
}

.c0:disabled {
  color: #BDBDBD;
  border-color: #BDBDBD;
  cursor: default;
}

<div>
  <button
    class="c0"
  >
    Button
  </button>
</div>
`;

exports[`<Button/> should render correctly when the type is outlined and the variant is valid 8`] = `
.c0 {
  background-color: transparent;
  border-radius: 24px;
  border-color: #000000;
  color: #000000;
  padding: 6px 15px;
  border-width: 2px;
  border-style: solid;
  font-size: 14px;
  font-family: Quicksand;
  font-weight: bold;
  outline: none;
  cursor: pointer;
}

.c0:disabled {
  color: #BDBDBD;
  border-color: #BDBDBD;
  cursor: default;
}

<div>
  <button
    class="c0"
  >
    Button
  </button>
</div>
`;

exports[`<Button/> should render correctly when the type is outlined and the variant is valid 9`] = `
.c0 {
  background-color: transparent;
  border-radius: 24px;
  border-color: #34333A;
  color: #34333A;
  padding: 6px 15px;
  border-width: 2px;
  border-style: solid;
  font-size: 14px;
  font-family: Quicksand;
  font-weight: bold;
  outline: none;
  cursor: pointer;
}

.c0:disabled {
  color: #BDBDBD;
  border-color: #BDBDBD;
  cursor: default;
}

<div>
  <button
    class="c0"
  >
    Button
  </button>
</div>
`;

exports[`<Button/> should render correctly when the type is outlined and the variant is valid 10`] = `
.c0 {
  background-color: transparent;
  border-radius: 24px;
  border-color: #80485b;
  color: #80485b;
  padding: 6px 15px;
  border-width: 2px;
  border-style: solid;
  font-size: 14px;
  font-family: Quicksand;
  font-weight: bold;
  outline: none;
  cursor: pointer;
}

.c0:disabled {
  color: #BDBDBD;
  border-color: #BDBDBD;
  cursor: default;
}

<div>
  <button
    class="c0"
  >
    Button
  </button>
</div>
`;

exports[`<Button/> should render correctly when the type is outlined and the variant is valid 11`] = `
.c0 {
  background-color: transparent;
  border-radius: 24px;
  border-color: #fff0de;
  color: #fff0de;
  padding: 6px 15px;
  border-width: 2px;
  border-style: solid;
  font-size: 14px;
  font-family: Quicksand;
  font-weight: bold;
  outline: none;
  cursor: pointer;
}

.c0:disabled {
  color: #BDBDBD;
  border-color: #BDBDBD;
  cursor: default;
}

<div>
  <button
    class="c0"
  >
    Button
  </button>
</div>
`;

exports[`<Button/> should render correctly when the type is raised and the variant is valid 1`] = `
.c0 {
  background-color: #F9A109;
  border-color: #F9A109;
  border-radius: 12px;
  color: #FFFFFF;
  padding: 20px;
  border-width: 2px;
  border-style: solid;
  font-size: 14px;
  font-family: Quicksand;
  font-weight: bold;
  outline: none;
  cursor: pointer;
}

.c0:disabled {
  background-color: #BDBDBD;
  border-color: #BDBDBD;
  cursor: default;
}

<div>
  <button
    class="c0"
  >
    Button
  </button>
</div>
`;

exports[`<Button/> should render correctly when the type is raised and the variant is valid 2`] = `
.c0 {
  background-color: #56CCF2;
  border-color: #56CCF2;
  border-radius: 12px;
  color: #FFFFFF;
  padding: 20px;
  border-width: 2px;
  border-style: solid;
  font-size: 14px;
  font-family: Quicksand;
  font-weight: bold;
  outline: none;
  cursor: pointer;
}

.c0:disabled {
  background-color: #BDBDBD;
  border-color: #BDBDBD;
  cursor: default;
}

<div>
  <button
    class="c0"
  >
    Button
  </button>
</div>
`;

exports[`<Button/> should render correctly when the type is raised and the variant is valid 3`] = `
.c0 {
  background-color: #EB5757;
  border-color: #EB5757;
  border-radius: 12px;
  color: #FFFFFF;
  padding: 20px;
  border-width: 2px;
  border-style: solid;
  font-size: 14px;
  font-family: Quicksand;
  font-weight: bold;
  outline: none;
  cursor: pointer;
}

.c0:disabled {
  background-color: #BDBDBD;
  border-color: #BDBDBD;
  cursor: default;
}

<div>
  <button
    class="c0"
  >
    Button
  </button>
</div>
`;

exports[`<Button/> should render correctly when the type is raised and the variant is valid 4`] = `
.c0 {
  background-color: #FFFFFF;
  border-color: #FFFFFF;
  border-radius: 12px;
  color: #FFFFFF;
  padding: 20px;
  border-width: 2px;
  border-style: solid;
  font-size: 14px;
  font-family: Quicksand;
  font-weight: bold;
  outline: none;
  cursor: pointer;
}

.c0:disabled {
  background-color: #BDBDBD;
  border-color: #BDBDBD;
  cursor: default;
}

<div>
  <button
    class="c0"
  >
    Button
  </button>
</div>
`;

exports[`<Button/> should render correctly when the type is raised and the variant is valid 5`] = `
.c0 {
  background-color: #E0E0E0;
  border-color: #E0E0E0;
  border-radius: 12px;
  color: #FFFFFF;
  padding: 20px;
  border-width: 2px;
  border-style: solid;
  font-size: 14px;
  font-family: Quicksand;
  font-weight: bold;
  outline: none;
  cursor: pointer;
}

.c0:disabled {
  background-color: #BDBDBD;
  border-color: #BDBDBD;
  cursor: default;
}

<div>
  <button
    class="c0"
  >
    Button
  </button>
</div>
`;

exports[`<Button/> should render correctly when the type is raised and the variant is valid 6`] = `
.c0 {
  background-color: #BDBDBD;
  border-color: #BDBDBD;
  border-radius: 12px;
  color: #FFFFFF;
  padding: 20px;
  border-width: 2px;
  border-style: solid;
  font-size: 14px;
  font-family: Quicksand;
  font-weight: bold;
  outline: none;
  cursor: pointer;
}

.c0:disabled {
  background-color: #BDBDBD;
  border-color: #BDBDBD;
  cursor: default;
}

<div>
  <button
    class="c0"
  >
    Button
  </button>
</div>
`;

exports[`<Button/> should render correctly when the type is raised and the variant is valid 7`] = `
.c0 {
  background-color: #828282;
  border-color: #828282;
  border-radius: 12px;
  color: #FFFFFF;
  padding: 20px;
  border-width: 2px;
  border-style: solid;
  font-size: 14px;
  font-family: Quicksand;
  font-weight: bold;
  outline: none;
  cursor: pointer;
}

.c0:disabled {
  background-color: #BDBDBD;
  border-color: #BDBDBD;
  cursor: default;
}

<div>
  <button
    class="c0"
  >
    Button
  </button>
</div>
`;

exports[`<Button/> should render correctly when the type is raised and the variant is valid 8`] = `
.c0 {
  background-color: #000000;
  border-color: #000000;
  border-radius: 12px;
  color: #FFFFFF;
  padding: 20px;
  border-width: 2px;
  border-style: solid;
  font-size: 14px;
  font-family: Quicksand;
  font-weight: bold;
  outline: none;
  cursor: pointer;
}

.c0:disabled {
  background-color: #BDBDBD;
  border-color: #BDBDBD;
  cursor: default;
}

<div>
  <button
    class="c0"
  >
    Button
  </button>
</div>
`;

exports[`<Button/> should render correctly when the type is raised and the variant is valid 9`] = `
.c0 {
  background-color: #34333A;
  border-color: #34333A;
  border-radius: 12px;
  color: #FFFFFF;
  padding: 20px;
  border-width: 2px;
  border-style: solid;
  font-size: 14px;
  font-family: Quicksand;
  font-weight: bold;
  outline: none;
  cursor: pointer;
}

.c0:disabled {
  background-color: #BDBDBD;
  border-color: #BDBDBD;
  cursor: default;
}

<div>
  <button
    class="c0"
  >
    Button
  </button>
</div>
`;

exports[`<Button/> should render correctly when the type is raised and the variant is valid 10`] = `
.c0 {
  background-color: #80485b;
  border-color: #80485b;
  border-radius: 12px;
  color: #FFFFFF;
  padding: 20px;
  border-width: 2px;
  border-style: solid;
  font-size: 14px;
  font-family: Quicksand;
  font-weight: bold;
  outline: none;
  cursor: pointer;
}

.c0:disabled {
  background-color: #BDBDBD;
  border-color: #BDBDBD;
  cursor: default;
}

<div>
  <button
    class="c0"
  >
    Button
  </button>
</div>
`;

exports[`<Button/> should render correctly when the type is raised and the variant is valid 11`] = `
.c0 {
  background-color: #fff0de;
  border-color: #fff0de;
  border-radius: 12px;
  color: #FFFFFF;
  padding: 20px;
  border-width: 2px;
  border-style: solid;
  font-size: 14px;
  font-family: Quicksand;
  font-weight: bold;
  outline: none;
  cursor: pointer;
}

.c0:disabled {
  background-color: #BDBDBD;
  border-color: #BDBDBD;
  cursor: default;
}

<div>
  <button
    class="c0"
  >
    Button
  </button>
</div>
`;
